File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.cpp
#include <stack>
#include <map>
using namespace std;


class Solution {
public:
    void dfs(TreeNode* node,int depth, int& maksDepth, int& sum) {
        if (!node) {
            return;
        }
        if (depth > maksDepth) {
            maksDepth = depth;
            sum = 0;
        }
        if (maksDepth == depth) {
            sum += node->val;
        }

        dfs(node->left,depth+1,maksDepth,sum);
        dfs(node->right,depth+1,maksDepth,sum);
    }
    int deepestLeavesSum(TreeNode* root) {
        int sum = 0;
        int maksDepth = 0;
        dfs(root,0,maksDepth,sum);
        return  sum;
    }
};

File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.java
class Solution {
        public int deepestLeavesSum(TreeNode root) {
            int[] maxDepth = new int[1];
            int[] sum = new int[1];
            solve(root, maxDepth,0,sum);
            return sum[0];
        }
        
        public void solve(TreeNode root,int[] maxDepth, int currDepth, int[] sum){
            if (root == null){
                return;
            }
            if (currDepth > maxDepth[0]){
                maxDepth[0] = currDepth;
                sum[0] = root.val;
            }else if (currDepth == maxDepth[0]){
                sum[0] += root.val;
            }
            solve(root.left,maxDepth,currDepth+1,sum);
            solve(root.right,maxDepth,currDepth+1,sum);
        }
    }